File: chapter_4/Employee/Employee.java
public class Employee {
    private String first_name;
    private String last_name;
    private double wage;

    public Employee(String first_name, String last_name, double wage) {
        this.first_name = first_name;
        this.last_name = last_name;
        this.wage = wage;
    }

    public String getFirstName() {
        return first_name;
    }

    public String getLastName() {
        return last_name;
    }

    public double getWage() {
        return wage;
    }
}

public class CommissionedEmployee extends Employee {
    private double commision;
    private int units;

    public CommissionedEmployee(String first_name, String last_name, double wage, double commision) {
       super(first_name, last_name, wage);
       this.commision = commision;
    }

    public double calculatePay() {
        return getWage() + (commision * units);
    }

    public void addSales(int units) {
        this.units += units;
    }

    public void resetSales() {
        this.units = 0;
    }
}

File: chapter_4/Employee/employee.py
class Employee:
    def __init__(self, first_name, last_name, wage):
        self.first_name = first_name
        self.last_name = last_name
        self.wage = wage


class CommisionedEmployee(Employee):
    units = 0

    def __init__(self, first_name, last_name, wage, commision):
        super().__init__(first_name, last_name, wage)
        self.commision = commision

    def calculate_pay(self):
        return self.wage + (self.commision * self.units)

    def add_sale(self, units):
        self.units += units

    def reset_sales(self):
        self.units = 0
